pkg/diff/sql_generator.go

```diff
@@ -1970,23 +1986,77 @@ func (a *attachPartitionSQLVertexGenerator) GetDeleteDependencies(_ schema.Table
 	return nil, nil
 }
 
+func buildForeignKeyConstraintDiff(fsg *foreignKeyConstraintSQLVertexGenerator, addedTablesByName map[string]schema.Table, old, new schema.ForeignKeyConstraint) (foreignKeyConstraintDiff, bool, error) {
```

Comment:
This aligns more with the new paradigm of sql vertex generators: the diff function should call into the sql vertex generator (if possible) to reduce duplicated logic.